api/Api/EnvironmentManager/Components/Rigidbody.cs

```diff
@@ -0,0 +1,207 @@
+            get => maxDepenetrationVelocity;
+            set {
+                maxDepenetrationVelocity = value; // TODO: Validate?
+                OnPropertyChanged();
+            }
+        }
+        internal CollisionDetectionMode collisionDetectionMode = CollisionDetectionMode.Discrete;
+        /// <summary>
+        /// 
+        /// </summary>
+        public CollisionDetectionMode CollisionDetectionMode {
+            get => collisionDetectionMode;
+            set {
+                collisionDetectionMode = value;
+                OnPropertyChanged();
+            }
+        }
+        internal RigidbodyConstraints constraints = RigidbodyConstraints.None;
+        public RigidbodyConstraints Constraints {
+            get => constraints;
+            set {
+                constraints = value;
+                OnPropertyChanged();
+            }
+        }
+
+        #endregion
+
+        internal List<(Vector3D Force, Vector3D Position, bool Relative, ForceMode Mode)> AdditionalForces
+            { get; private set; } = new List<(Vector3D Force, Vector3D Position, bool Relative, ForceMode Mode)>();
+        internal List<(Vector3D Torque, bool Relative, ForceMode Mode)> AdditionalTorques
+            { get; private set; } = new List<(Vector3D Torque, bool Relative, ForceMode Mode)>();
+
+        /// <summary>
+        /// Add force to body
+        /// </summary>
+        /// <param name="force">Force in Newtons * DeltaT</param>
+        /// <param name="relative"></param>
+        /// <param name="mode"></param>
+        public void AddForce(Vector3D force, bool relative = false, ForceMode mode = ForceMode.Force)
+        {
+            AdditionalForces.Add((force, Vector3D.NaN, relative, mode));
+            Changed = true;
+        }
+
+        public void AddForce(Vector3D force, Vector3D position, ForceMode mode = ForceMode.Force)
+        {
+            AdditionalForces.Add((force, position, false, mode));
+            Changed = true;
+        }
+
+        public void AddTorque(Vector3D torque, bool relative = false, ForceMode mode = ForceMode.Force)
+        {
+            AdditionalTorques.Add((torque, relative, mode));
+            Changed = true;
+        }
+
+        internal bool Changed { get; private set; } = true;
+        internal void ProcessedChanges() => Changed = false;
+
+        private TResult ConvertEnum<TResult>(object i) => (TResult)Enum.Parse(typeof(TResult), i.ToString(), true);
+
+        protected void OnPropertyChanged([CallerMemberName] string name = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
+        }
+    }
+
+    #region Enums
+
+    public enum CollisionDetectionMode
+    {
+        Discrete, Continuous, ContinuousDynamic,
+        ContinuousSpeculative
+    }
+
+    public enum ForceMode
+    {
+        Force = 0, Acceleration = 5,
+        Impulse = 1, VelocityChange = 2
+    }
+
+    public enum RigidbodyConstraints
```

Comment:
You may consider adding the `Flags` attribute to this enumeration which allows an enum value to contain many values. That way we could put multiple different constraints on things at once.